torchbenchmark/models/maml/__init__.py

```diff
@@ -0,0 +1,85 @@
+from argparse import Namespace
+import math
+import time
+from tqdm import tqdm
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from .meta import Meta
+
+
+import random
+import numpy as np
+from pathlib import Path
+
+torch.manual_seed(1337)
+random.seed(1337)
+np.random.seed(1337)
+torch.backends.cudnn.deterministic = True
+torch.backends.cudnn.benchmark = False
+
+class Model:
+    def __init__(self, device=None, jit=False):
+        self.device = device
+        self.jit = jit
+        root = str(Path(__file__).parent)
+        args = Namespace(**{
+            'n_way': 5,
+            'k_spt': 1,
+            'k_qry': 15,
+            'imgsz': 28,
+            'imgc': 1,
+            'task_num': 32,
+            'meta_lr': 1e-3,
+            'update_lr': 0.4,
+            'update_step': 5,
+            'update_step_test': 10
+        })
+        config = [
+            ('conv2d', [64, 1, 3, 3, 2, 0]),
+            ('relu', [True]),
+            ('bn', [64]),
+            ('conv2d', [64, 64, 3, 3, 2, 0]),
+            ('relu', [True]),
+            ('bn', [64]),
+            ('conv2d', [64, 64, 3, 3, 2, 0]),
+            ('relu', [True]),
+            ('bn', [64]),
+            ('conv2d', [64, 64, 2, 2, 1, 0]),
+            ('relu', [True]),
+            ('bn', [64]),
+            ('flatten', []),
+            ('linear', [args.n_way, 64])
+        ]
+        self.module = Meta(args, config).to(device)
+        self.example_inputs = torch.load(f'{root}/batch.pt')
+        self.example_inputs = tuple([torch.from_numpy(i).to(self.device) for i in self.example_inputs])
+
+        
+    def get_module(self):
+        if self.jit:
+            raise NotImplementedError()
+        return self.module, self.example_inputs
+
+    def eval(self, niter=1):
+        if self.jit:
+            raise NotImplementedError()
+        self.module.eval()
+        for _ in range(niter):
+            self.module.finetunning(*[i[0] for i in self.example_inputs])
```

Comment:
what's the structure of the example inputs? might be better to do the list comprehension outside and save the result as a separate inputs member, since 'eval' will get called multiple times by the benchmark driver and you'll end up timing that part.

also, finetunning is a funny typo right? but not your code, their code?

Reply:
> what's the structure of the example inputs?

It's just 5 inputs, so I thought it was fine.

> also, finetunning is a funny typo right? but not your code, their code?

Yep :) I find that some researchers have a very very annoying habit of making a typo somewhere and just keeping it throughout their entire code.